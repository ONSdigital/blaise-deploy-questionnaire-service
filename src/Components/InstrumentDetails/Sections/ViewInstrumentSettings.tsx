import React, {ReactElement, useEffect, useState} from "react";
import {getInstrumentModes, getInstrumentSettings} from "../../../utilities/http";
import {Instrument} from "../../../../Interfaces";
import {ONSPanel} from "blaise-design-system-react-components";
import {InstrumentSettings} from "blaise-api-node-client";
import {formatText} from "../../../utilities/TextFormatting/TextFormatting";

interface Props {
    instrument: Instrument;
}

const ViewInstrumentSettings = ({instrument}: Props): ReactElement => {
    const [setting, setSetting] = useState<InstrumentSettings>();
    const [mode, setMode] = useState<string>();
    const [errored, setErrored] = useState<boolean>(false);

    useEffect(() => {
        getInstrumentModes(instrument.name)
            .then((data) => {
                console.log(data);
                if (data === null) {
                    setErrored(true);
                    return;
                }
                if (data.includes("CATI")) {
                    setMode("CATI");
                }
                if (data.length > 1) {
                    setMode("Mixed");
                }
                console.log(mode);
            });

        getInstrumentSettings(instrument.name)
            .then((data) => {
                if (data === null) {
                    setErrored(true);
                    return;
                }
                console.log(data);
                const setting = data.find(x => x.type === "StrictInterviewing");
                if (setting !== undefined) {
                    setSetting(setting);
                }
            });
    }, []);

    const validCatiModeSettings = [
        {
            "type": "StrictInterviewing",
            "saveSessionOnTimeout": true,
            "saveSessionOnQuit": true,
            "deleteSessionOnTimeout": "any",
            "deleteSessionOnQuit": "any",
            "applyRecordLocking": false
        }
    ];

    const validMixedModeSettings =
        {
            "type": "StrictInterviewing",
            "saveSessionOnTimeout": true,
            "saveSessionOnQuit": true,
            "deleteSessionOnTimeout": true,
            "deleteSessionOnQuit": true,
            "applyRecordLocking": true
        };

    // function shallowEqual(expectedSetting: object, actualSetting: object) {
    //     const expectedKeys = Object.keys(expectedSetting);
    //     const actualKeys = Object.keys(actualSetting);
    //
    //     if (expectedKeys.length !== actualKeys.length) {
    //         return false;
    //     }
    //     for (let key of expectedKeys) {
    //         if (expectedSetting[key] !== actualSetting[key]) {
    //             return false;
    //         }
    //     }
    // };

    useEffect(() => {
        if (mode === "CATI") {
            // Rules for CATI ONLY questionnaires:
            //
            // saveSessionOnTimeout: true
            // saveSessionOnQuit: true
            // deleteSessionOnTimeout: any
            // deleteSessionOnQuit: any
            // applyRecordLocking: any
        }
        if (mode == "Mixed") {
            if (setting === null) {
                console.log("Setting was null");
                return;
            }

            console.log(setting);
            console.log(validMixedModeSettings);
            if (setting !== validMixedModeSettings) {
                console.log("computer says no");
            }
            // Rules for Mixed mode questionnaires:
            //
            // saveSessionOnTimeout: true
            // saveSessionOnQuit: true
            // deleteSessionOnTimeout: true
            // deleteSessionOnQuit: true
            // applyRecordLocking: true
        }
    }, [setting]);

    if (errored) {
        return (
            <>
                <ONSPanel status={"error"}>
                    <p>Failed to get questionnaire settings</p>
                </ONSPanel>
            </>

        );
    }

    function convertJsonToTable(object: any) {
        const elementList: ReactElement[] = [];
        const entries: [string, (string | null | number)][] = Object.entries(object);
        for (const [field, data] of entries) {
            elementList.push(
                <tbody className="summary__item" key={field}>
                <tr className="summary__row summary__row--has-values">
                    <td className="summary__item-title">
                        <div className="summary__item--text">
                            {formatText(field)}
                        </div>
                    </td>
                    <td className="summary__values" colSpan={2}>
                        {(typeof data === "boolean") ? (data ? "True" : "False") : data}
                    </td>
                </tr>
                </tbody>
            );
        }
        return elementList.map((element => {
            return element;
        }));
    }

    if (setting) {
        return (
            <div className="summary">
                <div className="summary__group">
                    <h2>Questionnaire settings</h2>
                    <table id="report-table" className="summary__items u-mt-s">
                        {
                            convertJsonToTable(setting)
                        }
                    </table>
                </div>
            </div>
        );
    }


    return (
        <div>
            <p>Loading...</p>
        </div>
    );
};

export default ViewInstrumentSettings;
