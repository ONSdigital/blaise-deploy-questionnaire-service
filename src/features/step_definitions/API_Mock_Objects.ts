import { Survey } from "../../../Interfaces";

export const survey_list_with_OPN_and_LMS_with_one_active_instrument_each: Survey[] = [
    {
        survey: "OPN",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "July 2020",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2007T",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            }
        ]
    },
    {
        survey: "LMS",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "Unknown",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/LMSSurvey01?LayoutSet=CATI-Interviewer_Large",
                name: "LMSSurvey01",
                serverParkName: "LocalDevelopment",
                surveyTLA: "LMS",
                surveyDays: []
            }
        ]
    }
];

export const survey_list_with_OPN_with_one_active_instrument: Survey[] = [
    {
        survey: "OPN",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "July 2020",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2007T",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            }
        ]
    }
];

export const survey_list_with_OPN_with_two_active_instruments: Survey[] = [
    {
        survey: "OPN",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "July 2020",
                expired: false,
                installDate: "2020-12-10T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2007T",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            },
            {
                activeToday: true,
                fieldPeriod: "April 2020",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2004A?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2004A",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            }
        ]
    }
];

export const survey_list_with_OPN_with_three_active_instruments: Survey[] = [
    {
        survey: "OPN",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "July 2020",
                expired: false,
                installDate: "2020-12-10T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2007T",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            },
            {
                activeToday: true,
                fieldPeriod: "January 2021",
                expired: false,
                installDate: "2020-12-12T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2101A?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2101A",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            },
            {
                activeToday: true,
                fieldPeriod: "April 2020",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2004A?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2004A",
                serverParkName: "LocalDevelopment",
                surveyTLA: "OPN",
                surveyDays: []
            },
        ]
    }
];
