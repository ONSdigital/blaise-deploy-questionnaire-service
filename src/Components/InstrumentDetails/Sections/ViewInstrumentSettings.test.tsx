import flushPromises from "../../../tests/utils";
import {cleanup, render, screen, waitFor} from "@testing-library/react";
import {act} from "react-dom/test-utils";
import React from "react";
import ViewInstrumentSettings from "./ViewInstrumentSettings";
import {opnInstrument} from "../../../features/step_definitions/API_Mock_Objects";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import {mock_fetch_requests} from "../../../features/step_definitions/functions";

const viewInstrumentSettingsFailedMessage = /Failed to get questionnaire settings/i;
const InstrumentSettingsMockList = [
    {
        "type": "StrictInterviewing",
        "saveSessionOnTimeout": false,
        "saveSessionOnQuit": true,
        "deleteSessionOnTimeout": false,
        "deleteSessionOnQuit": false,
        "sessionTimeout": 15,
        "applyRecordLocking": false
    }
];


describe("Given the API successfully loads the instrument mode and settings for CATI only mode", () => {

    const mock_server_responses = (url: string) => {
        console.log(url);
        if (url.includes("/modes")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve(["CATI"]),
            });
        }
        if (url.includes("/settings")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve(InstrumentSettingsMockList),
            });
        }
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve([]),
        });
    };

    beforeAll(() => {
        mock_fetch_requests(mock_server_responses);
    });

    it("matches Snapshot for the view Instrument Settings page", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("should render correctly", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(/Questionnaire settings/i)).toBeDefined();
            expect(screen.getByText(/Type/i)).toBeDefined();
            expect(screen.getByText(/SaveSessionOnQuit/i)).toBeDefined();
            expect(screen.getByText(/SessionTimeout/i)).toBeDefined();
        });
    });

    it("should highlight all invalid fields", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(/SaveSessionOnTimeout should be True/i)).toBeDefined();
        });
    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});

describe("Given the API successfully loads the instrument mode and settings for mixed mode", () => {

    const mock_server_responses = (url: string) => {
        console.log(url);
        if (url.includes("/modes")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve(["CATI", "CAWI"]),
            });
        }
        if (url.includes("/settings")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve(InstrumentSettingsMockList),
            });
        }
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve([]),
        });
    };

    beforeAll(() => {
        mock_fetch_requests(mock_server_responses);
    });

    it("matches Snapshot for the view Instrument Settings page", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("should render correctly", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(/Questionnaire settings/i)).toBeDefined();
            expect(screen.getByText(/Type/i)).toBeDefined();
            expect(screen.getByText(/SaveSessionOnQuit/i)).toBeDefined();
            expect(screen.getByText(/SessionTimeout/i)).toBeDefined();
        });
    });

    it("should highlight all invalid fields", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(/SaveSessionOnTimeout should be True/i)).toBeDefined();
            expect(screen.getByText(/DeleteSessionOnTimeout should be True/i)).toBeDefined();
            expect(screen.getByText(/DeleteSessionOnQuit should be True/i)).toBeDefined();
            expect(screen.getByText(/ApplyRecordLocking should be True/i)).toBeDefined();
        });
    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});

describe("Given the API fails to load the instrument mode and settings", () => {

    const mock_server_responses = (url: string) => {
        console.log(url);
        if (url.includes("/modes")) {
            return Promise.resolve({
                status: 500,
                json: () => Promise.resolve({}),
            });
        }
        if (url.includes("/settings")) {
            return Promise.resolve({
                status: 500,
                json: () => Promise.resolve({}),
            });
        }
        return Promise.resolve({
            status: 500,
            json: () => Promise.resolve({}),
        });
    };

    beforeAll(() => {
        mock_fetch_requests(mock_server_responses);
    });

    it("should display an error message when it fails to load the Instrument Modes", async () => {
        render(
            <ViewInstrumentSettings instrument={opnInstrument}/>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(viewInstrumentSettingsFailedMessage)).toBeDefined();
        });
    });

    it("should display an error message when it fails to load the Instrument Settings", async () => {
        render(
            <ViewInstrumentSettings instrument={opnInstrument}
            />
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(viewInstrumentSettingsFailedMessage)).toBeDefined();
        });
    });
});

describe("Given the API returns an empty list for instrument mode and/or settings", () => {

    const mock_server_responses = (url: string) => {
        console.log(url);
        if (url.includes("/modes")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve([]),
            });
        }
        if (url.includes("/settings")) {
            return Promise.resolve({
                status: 200,
                json: () => Promise.resolve([]),
            });
        }
        return Promise.resolve({
            status: 200,
            json: () => Promise.resolve([]),
        });
    };

    beforeAll(() => {
        mock_fetch_requests(mock_server_responses);
    });

    it("it should render an error message", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <ViewInstrumentSettings instrument={opnInstrument}/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(screen.getByText(viewInstrumentSettingsFailedMessage)).toBeDefined();
        });
    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});
