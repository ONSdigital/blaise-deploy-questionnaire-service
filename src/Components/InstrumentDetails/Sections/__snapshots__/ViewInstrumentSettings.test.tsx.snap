// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the API successfully loads the instrument mode and settings for CATI only mode matches Snapshot for the view Instrument Settings page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="summary"
      >
        <div
          class="summary__group"
        >
          <h2>
            Questionnaire settings
          </h2>
          <table
            class="summary__items u-mt-s"
            id="report-table"
          >
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Type
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  StrictInterviewing
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item summary__item--error"
            >
              <tr
                class="summary__row"
              >
                <th
                  class="summary__row-title u-fs-r"
                  colspan="3"
                >
                  SaveSessionOnTimeout
                   should be 
                  True
                </th>
              </tr>
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SaveSessionOnTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SaveSessionOnQuit
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  True
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    DeleteSessionOnTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    DeleteSessionOnQuit
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SessionTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  15
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    ApplyRecordLocking
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="summary"
    >
      <div
        class="summary__group"
      >
        <h2>
          Questionnaire settings
        </h2>
        <table
          class="summary__items u-mt-s"
          id="report-table"
        >
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Type
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                StrictInterviewing
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item summary__item--error"
          >
            <tr
              class="summary__row"
            >
              <th
                class="summary__row-title u-fs-r"
                colspan="3"
              >
                SaveSessionOnTimeout
                 should be 
                True
              </th>
            </tr>
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SaveSessionOnTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SaveSessionOnQuit
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                True
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  DeleteSessionOnTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  DeleteSessionOnQuit
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SessionTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                15
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  ApplyRecordLocking
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Given the API successfully loads the instrument mode and settings for mixed mode matches Snapshot for the view Instrument Settings page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="summary"
      >
        <div
          class="summary__group"
        >
          <h2>
            Questionnaire settings
          </h2>
          <table
            class="summary__items u-mt-s"
            id="report-table"
          >
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    Type
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  StrictInterviewing
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item summary__item--error"
            >
              <tr
                class="summary__row"
              >
                <th
                  class="summary__row-title u-fs-r"
                  colspan="3"
                >
                  SaveSessionOnTimeout
                   should be 
                  True
                </th>
              </tr>
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SaveSessionOnTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SaveSessionOnQuit
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  True
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item summary__item--error"
            >
              <tr
                class="summary__row"
              >
                <th
                  class="summary__row-title u-fs-r"
                  colspan="3"
                >
                  DeleteSessionOnTimeout
                   should be 
                  True
                </th>
              </tr>
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    DeleteSessionOnTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item summary__item--error"
            >
              <tr
                class="summary__row"
              >
                <th
                  class="summary__row-title u-fs-r"
                  colspan="3"
                >
                  DeleteSessionOnQuit
                   should be 
                  True
                </th>
              </tr>
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    DeleteSessionOnQuit
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item "
            >
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    SessionTimeout
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  15
                </td>
              </tr>
            </tbody>
            <tbody
              class="summary__item summary__item--error"
            >
              <tr
                class="summary__row"
              >
                <th
                  class="summary__row-title u-fs-r"
                  colspan="3"
                >
                  ApplyRecordLocking
                   should be 
                  True
                </th>
              </tr>
              <tr
                class="summary__row summary__row--has-values"
              >
                <td
                  class="summary__item-title"
                >
                  <div
                    class="summary__item--text"
                  >
                    ApplyRecordLocking
                  </div>
                </td>
                <td
                  class="summary__values"
                  colspan="2"
                >
                  False
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="summary"
    >
      <div
        class="summary__group"
      >
        <h2>
          Questionnaire settings
        </h2>
        <table
          class="summary__items u-mt-s"
          id="report-table"
        >
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  Type
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                StrictInterviewing
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item summary__item--error"
          >
            <tr
              class="summary__row"
            >
              <th
                class="summary__row-title u-fs-r"
                colspan="3"
              >
                SaveSessionOnTimeout
                 should be 
                True
              </th>
            </tr>
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SaveSessionOnTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SaveSessionOnQuit
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                True
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item summary__item--error"
          >
            <tr
              class="summary__row"
            >
              <th
                class="summary__row-title u-fs-r"
                colspan="3"
              >
                DeleteSessionOnTimeout
                 should be 
                True
              </th>
            </tr>
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  DeleteSessionOnTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item summary__item--error"
          >
            <tr
              class="summary__row"
            >
              <th
                class="summary__row-title u-fs-r"
                colspan="3"
              >
                DeleteSessionOnQuit
                 should be 
                True
              </th>
            </tr>
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  DeleteSessionOnQuit
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item "
          >
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  SessionTimeout
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                15
              </td>
            </tr>
          </tbody>
          <tbody
            class="summary__item summary__item--error"
          >
            <tr
              class="summary__row"
            >
              <th
                class="summary__row-title u-fs-r"
                colspan="3"
              >
                ApplyRecordLocking
                 should be 
                True
              </th>
            </tr>
            <tr
              class="summary__row summary__row--has-values"
            >
              <td
                class="summary__item-title"
              >
                <div
                  class="summary__item--text"
                >
                  ApplyRecordLocking
                </div>
              </td>
              <td
                class="summary__values"
                colspan="2"
              >
                False
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
